TemplateProject/fastlane/Fastfile

```diff
@@ -0,0 +1,95 @@
+fastlane_version "2.44.1"
+fastlane_require "semantic"
+
+xcodeproj = "./ios/TemplateProject.xcodeproj"
+build_gradle = "../android/app/build.gradle"
+
+desc "Bump the iOS and Android version with a commit and tag."
+lane :bump_version do |options|
+  type = options.fetch(:type)
+  android_incrementor = AndroidVersionIncrementer.new(build_gradle)
+
+  case type
+  when "major", "minor", "patch"
+    increment_version_number(bump_type: type, xcodeproj: xcodeproj)
+    android_incrementor.increment_version_name!(type)
+  else
+    raise "Invalid bump type '#{type}'"
+  end
+
+  commit_tag_and_push_version_bump(xcodeproj: xcodeproj)
+end
+
+def commit_tag_and_push_version_bump(xcodeproj:)
+  build_number = get_build_number(xcodeproj: xcodeproj)
+  version_number = get_version_number(xcodeproj: xcodeproj)
+  git_commit(
+    path: [
+      "android/app/build.gradle",
+      "ios/TemplateProject.xcodeproj/project.pbxproj",
+      "ios/TemplateProject/Info.plist",
+      "ios/TemplateProjectTests/Info.plist",
+    ],
+    message: "Bump version to #{version_number} (#{build_number})",
+  )
+  add_git_tag(tag: "v#{version_number}+#{build_number}")
+  push_to_git_remote
+end
+
+platform :ios do
+  desc "Bulid the app for distribution"
+  lane :build do
+    gym(project: xcodeproj, scheme: "TemplateProject", include_bitcode: true)
+  end
+
+  desc "Submit a new Beta Build to Apple TestFlight"
+  lane :beta do
+    build
+    pilot
+  end
+
+  desc "Create a new version on the App Store"
+  desc "This assumes the build on TestFlight is the same version"
+  lane :release do
+    deliver(force: true, skip_binary_upload: true)
+  end
+end
+
+platform :android do
+  desc "Bulid the app for release"
+  lane :build do
+    gradle(task: "clean", project_dir: "android/")
+    gradle(task: "assembleRelease", project_dir: "android/")
+  end
+end
+
+class AndroidVersionIncrementer
```

Comment:
I extracted these out into custom actions, which ended up being really nice. It seems like it would make them easier to share across template repos in the future.